### PR TITLE
OTA-1274,OTA-1277: upgrade status: use a table to represent node pools

### PR DESCRIPTION
**Pull request by hongkailiu**
https://issues.redhat.com/browse/OTA-1277

### CODE REVIEW

File: pkg/cli/admin/upgrade/status/examples/4.14.1-degraded.detailed-output
Comment (hongkailiu) [inline pkg/cli/admin/upgrade/status/examples/4.14.1-degraded.detailed-output:15]: I was following the card but ... do we want to use all-capital letters for the table header? (to be aligned with the "Worker Pool Nodes" table below?

Comment (petr-muller) [reply pkg/cli/admin/upgrade/status/examples/4.14.1-degraded.detailed-output:16]: yeah, let's all-cap for consistency

File: pkg/cli/admin/upgrade/status/workerpool.go
Comment (petr-muller) [inline pkg/cli/admin/upgrade/status/workerpool.go:492]:
```suggestion
		fmt.Fprintf(w, "\nWorker Pool Nodes: %s", pool.Name)
```

Now it does not make sense to try being smart about the plural (L494-496), let's just make the title `Worker Pool Nodes` always

File: pkg/cli/admin/upgrade/status/examples/4.15.0-ec2-b02-cos-not-annotated.detailed-output
Comment (petr-muller) [inline pkg/cli/admin/upgrade/status/examples/4.15.0-ec2-b02-cos-not-annotated.detailed-output:18]: Given we link to the OTA-1277 card can we handle the empty case nicer? Assessment `Completed` is fine I guess, but we could say `Scaled to zero`, `Downscaled` or `Empty` instead. Let's get rid of the `NaN` too. We could also try pruning the `Status` to just `0 Total` but maybe only for this case.

File: pkg/cli/admin/upgrade/status/examples/4.15.0-ec2-b02-cos-not-annotated.detailed-output
Comment (petr-muller) [inline pkg/cli/admin/upgrade/status/examples/4.15.0-ec2-b02-cos-not-annotated.detailed-output:27]: Let's not show this for empty pool.

**Review by petr-muller — CHANGES_REQUESTED** [review]
Looks good, but it seems more like [OTA-1274](https://issues.redhat.com/browse/OTA-1274) than [OTA-1277](https://issues.redhat.com/browse/OTA-1277) in the summary - either link it to OTA-1274 or address my nits about empty fields and let's close both (I do not have a preference).

**Comment by hongkailiu** [comment]
> Looks good, but it seems more like [OTA-1274](https://issues.redhat.com/browse/OTA-1274) than [OTA-1277](https://issues.redhat.com/browse/OTA-1277) in the summary - either link it to OTA-1274 or address my nits about empty fields and let's close both (I do not have a preference).

I linked 1274 by mistake. But I DID have the intention of addressing it with this PR.

File: pkg/cli/admin/upgrade/status/workerpool.go
Comment (petr-muller) [inline pkg/cli/admin/upgrade/status/workerpool.go:499]:
```suggestion
		fmt.Fprintf(w, "\nControl Plane Nodes")
```

Needs to happen too after we removed the plural smartassery

**Comment by petr-muller** [comment]
/retitle OTA-1274,OTA-1277: upgrade status: use a table to represent node pools

**Review by petr-muller — APPROVED** [review]
/label acknowledge-critical-fixes
/hold

Code LGTM
We're behind a feature flag so adding the TRT throttle label
Adding a /hold for @evakhoni , pls unhold after all required pre-merge testing is complete

**Comment by evakhoni** [comment]
> /label acknowledge-critical-fixes /hold
> 
> Code LGTM We're behind a feature flag so adding the TRT throttle label Adding a /hold for @evakhoni , pls unhold after all required pre-merge testing is complete

thanks @petr-muller! I actually already verified in [OTA-1274](https://issues.redhat.com/browse/OTA-1274?focusedId=25136832&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-25136832) so from my side we can  
/unhold  
feel free to hold again if any more verification is required.

**Pull request merged**
